src/cli/cli.cpp

```diff
@@ -4850,6 +4847,7 @@ otError Interpreter::ProcessNetworkDiagnostic(uint8_t aArgsLength, char *aArgs[]
 
     if (strcmp(aArgs[0], "get") == 0)
     {
+        otThreadSetReceiveDiagnosticGetCallback(mInstance, HandleDiagnosticGetResponse, this);
```

Comment:
Should we change the API to pass the callback as a part of `otThreadSendDiagnosticGet()`?

Reply:
SGTM, I changed the API accordingly, please review.